crates/engine/tree/src/tree/mod.rs

```diff
@@ -1194,7 +1194,18 @@ where
             return Ok(valid_outcome(state.head_block_hash))
         }
 
-        // 3. check if the head is already part of the canonical chain
+        // 3. Check if forkchoice state points to a valid ancestor of the canonical head.
+        if self.is_valid_canonical_ancestor(state.head_block_hash)? {
+            trace!(
+                target: "engine::tree",
+                head=?state.head_block_hash,
+                "fcu head block is a valid ancestor of the canonical head, skipping update"
+            );
+
+            return Ok(valid_outcome(state.head_block_hash));
+        }
```

Comment:
tis basically the same as 4 now, so we can remove this entirely.

and only need to swap 2,3